src/app/import/components/helpers/upload-helper.ts

```diff
@@ -0,0 +1,72 @@
+import { Router } from "@angular/router";
+import { timer } from "rxjs";
+import { ExistingProjectUploadHelper, LookupListsUploadHelper, RecordAddUploadHelper, RecordNewUploadHelper } from "./upload-specific";
+import { UploadFileType } from "./upload-types";
+import { UploadComponent } from "../upload/upload.component";
+
+export class UploadHelper {
+    baseComponent: UploadComponent;
+    recordNewUploadHelper: RecordNewUploadHelper;
+    recordAddUploadHelper: RecordAddUploadHelper;
+    existingProjectUploadHelper: ExistingProjectUploadHelper;
+    lookupListsUploadHelper: LookupListsUploadHelper;
+
+    constructor(baseComponent?: UploadComponent, recordNewUploadHelper?: RecordNewUploadHelper, recordAddUploadHelper?: RecordAddUploadHelper, existingProjectUploadHelper?: ExistingProjectUploadHelper, lookupListsUploadHelper?: LookupListsUploadHelper) {
+        this.baseComponent = baseComponent;
+        this.recordNewUploadHelper = recordNewUploadHelper;
+        this.recordAddUploadHelper = recordAddUploadHelper;
+        this.existingProjectUploadHelper = existingProjectUploadHelper;
+        this.lookupListsUploadHelper = lookupListsUploadHelper;
+    }
+
+    upload(file?: File): void {
+        if (file) this.baseComponent.file = file;
+        switch (this.baseComponent.uploadFileType) {
+            case UploadFileType.RECORDS_NEW:
+                this.recordNewUploadHelper.doUpload();
+                break;
+            case UploadFileType.RECORDS_ADD:
+                this.recordAddUploadHelper.doUpload();
+                break;
+            case UploadFileType.PROJECT:
+                this.existingProjectUploadHelper.doUpload();
+                break;
+            case UploadFileType.KNOWLEDGE_BASE:
+                this.lookupListsUploadHelper.doUpload();
+                break;
+        }
+    }
+
+    getFinalFileName(projectId: string, uploadFileType: UploadFileType, knowledgeBaseId?: string): string {
```

Comment:
the get method sets project id, an execute & a resubscirption? if these are necessary steps the calling methods should do that but not the file name getter => the next dev who tries to get the file name will be irritated